File: shared/src/commonMain/kotlin/pe/ecolecta/domain/VehiculoInvalidoException.kt
package pe.ecolecta.domain

sealed class VehiculoInvalidoException(mensaje: String) : Exception(mensaje) {
    data object NombreVacio : VehiculoInvalidoException("El nombre del vehículo es obligatorio.")
    data object PlacaVacia : VehiculoInvalidoException("La placa es obligatoria.")
    data object PlacaDuplicada : VehiculoInvalidoException("Ya existe un vehículo con esa placa.")
}

File: web/app/Domain/Vehiculos/Exceptions/VehiculoInvalidoException.php
<?php

namespace App\Domain\Vehiculos\Exceptions;

use DomainException;

final class VehiculoInvalidoException extends DomainException
{
    public static function nombreVacio(): self
    {
        return new self('El nombre del vehículo es obligatorio.');
    }

    public static function placaVacia(): self
    {
        return new self('La placa es obligatoria.');
    }

    public static function placaDuplicada(): self
    {
        return new self('Ya existe un vehículo con esa placa.');
    }
}
